src/transformStateWithClones.js

```diff
@@ -8,6 +8,48 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const results = [];
+  const stateClone = {};
+
+  Object.assign(stateClone, state);
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        {
+          const { extraData } = action;
+
+          Object.assign(stateClone, extraData);
+
+          results.push({ ...stateClone });
```

Comment:
To avoid repeating yourself, you should push a copy of the object to the array at the end of the loop cycle but outside of the `switch` block, after the current action is processed. This will make the code cleaner and follow the DRY principle. [DONT REPEAT YOURSELF]